tests/IntegrationTest.php

```diff
@@ -228,8 +228,9 @@ public function betweenOccurrences()
 
         $result = $criteria->build();
 
-        $this->assertTrue(is_array($result));
-        $this->assertTrue(count($result) >= 3 && count($result) <= 5);
```

Comment:
To reduce the logical/complicated condition in test, I think we should split two assertions from this logical/complex assertion.